nx/include/switch/services/btdrv.h

```diff
@@ -14,221 +14,291 @@
 
         struct {
-            u8 name[0xF9];               ///< Device name, NUL-terminated string.
-            BtdrvAddress addr;           ///< Device address.
-            u8 reserved_xFF[0x10];       ///< Reserved
-            u8 class_of_device[0x3];     ///< Class of Device.
-            u8 unk_x112[0x4];            ///< Set to fixed value u32 0x1.
-            u8 reserved_x116[0xFA];      ///< Reserved
-            u8 reserved_x210[0x5C];      ///< Reserved
-            u8 name2[0xF9];              ///< Device name, NUL-terminated string. Same as name above, except starting at index 1.
-            u8 rssi[0x4];                ///< s32 RSSI
-            u8 name3[0x4];               ///< Two bytes which are the same as name[11-12].
-            u8 reserved_x36D[0x10];      ///< Reserved
-        } inquiry_device;                ///< ::BtdrvEventType_InquiryDevice
+            union {
+                struct {
+                    char name[0xF9];                        ///< Device name, NUL-terminated string.
+                    BtdrvAddress addr;                      ///< Device address.
+                    u8 reserved_xFF[0x10];                  ///< Reserved
+                    BtdrvClassOfDevice class_of_device;     ///< Class of Device.
+                    u8 unk_x112[0x4];                       ///< Set to fixed value u32 0x1.
+                    u8 reserved_x116[0xFA];                 ///< Reserved
+                    u8 reserved_x210[0x5C];                 ///< Reserved
+                    char name2[0xF9];                       ///< Device name, NUL-terminated string. Same as name above, except starting at index 1.
+                    u8 rssi[0x4];                           ///< s32 RSSI
+                    u8 name3[0x4];                          ///< Two bytes which are the same as name[11-12].
+                    u8 reserved_x36D[0x10];                 ///< Reserved
+                } v1;                                       ///< [1.0.0-11.0.1]
+
+                struct {
+                    BtdrvAddress addr;                      ///< Device address.
+                    char name[0xF9];                        ///< Device name, NUL-terminated string.
+                    BtdrvClassOfDevice class_of_device;     ///< Class of Device.
+                    u8 reserved[0x6];                       ///< Reserved
+                } v12;                                      ///< [12.0.0+]
+            };
+        } inquiry_device;                                   ///< ::BtdrvEventType_InquiryDevice
+
+        struct {
+            union {
+                struct {
+                    BtdrvInquiryStatus status;              ///< \ref BtdrvInquiryStatus
+                } v1;                                       ///< [1.0.0-11.0.1]
+
+                struct {
+                    BtdrvInquiryStatus status;              ///< \ref BtdrvInquiryStatus
```

Comment:
This needs to be forcible u8 because an enum is normally int-sized.

Reply:
Good catch